src/wallet/rpcwallet.cpp

```diff
@@ -1164,6 +1175,10 @@ UniValue ListReceived(const UniValue& params, bool fByAccounts)
             if (!ExtractDestination(txout.scriptPubKey, address))
                 continue;
 
+            if (fFilterAddress && !(filterAddress == address)) {
```

Comment:
Not the same only in the sense that the `!=` operator may not be defined, right? Semantically they're identical. And yeah, I see now. `CNoDestination` has no `operator!=`. Ah well.